contracts/interfaces/IPaymaster.sol

```diff
@@ -34,65 +36,53 @@ interface IPaymaster {
     );
 
     /**
-     * Called by Relay (and RelayHub), to validate if this recipient accepts this call.
+     * Called by Relay (and RelayHub), to validate if the paymaster accepts paying this call.
      * revert to signal the paymaster will NOT pay for this call.
-     * Note: Accepting this call means paying for the tx whether the relayed call reverted or not.
+     *
+     * MUST be protected with relayHubOnly() in case it modifies state.
+     *
+     * Note that revert by the Forwrader (either on nonce or signature) will also reject the call.
+     * a paymaster may also set "revertOnRecipientRevert" to signal that revert by the recipient
+     * contract should also be rejected. In this case, it means the Paymaster trust the recipient
+     * to reject fast: both preRelayedCall, forwarder check and receipient checks must fit into
+     * the GasLimits.commitmentGasLimit, otherwise the TX is paid by the Paymaster.
      *  @param relayRequest - the full relay request structure
-     *  @param signature - user's EIP712-compatible signature of the {@link relayRequest}
      *  @param approvalData - extra dapp-specific data (e.g. signature from trusted party)
      *  @param maxPossibleGas - based on values returned from {@link getGasLimits},
      *         the RelayHub will calculate the maximum possible amount of gas the user may be charged for.
      *         In order to convert this value to wei, the Paymaster has to call "relayHub.calculateCharge()"
-     *  @return a context to be passed to preRelayedCall and postRelayedCall.
+     *  return:
+     *      a context to be passed to postRelayedCall
+     *      revertOnRecipientRevert - TRUE if paymaster want to reject the TX if the recipient reverts.
+     *          FALSE means that rejects by the recipient will be completed on chain, and paid by the paymaster.
+     *          (note that in this case, the preRelayedCall and postRelayedCall are not reverted).
```

Comment:
why not stick to `@return` natspec syntax?

Reply:
because @return doesn't work for me when returning a tupple.